packages/mui-material/src/AvatarGroup/AvatarGroup.js

```diff
@@ -98,7 +99,13 @@ const AvatarGroup = React.forwardRef(function AvatarGroup(inProps, ref) {
     return React.isValidElement(child);
   });
 
-  const extraAvatars = children.length > clampedMax ? children.length - clampedMax + 1 : 0;
+  const totalAvatars = total === undefined || total < 0 ? children.length : total;
+  const clampedChildren = totalAvatars < children.length ? totalAvatars : children.length;
+
+  const shownAvatars =
+    totalAvatars <= clampedMax || clampedChildren < clampedMax ? clampedChildren : clampedMax - 1;
+
+  const extraAvatars = totalAvatars - shownAvatars;
```

Comment:
Can you remove the `clampedChildren`? it makes the code very hard to understand and we need to document it.

```suggestion
  const totalAvatars = total || children.length;
  const extraAvatars = totalAvatars > clampedMax ? totalAvatars - clampedMax + 1 : 0;
```

Reply:
@siriwatknp I removed `clampedChildren` and refactored the code into something closer to your suggestion.

A problem arises when there are less child avatars than the `clampedMax`. In summary, we need to take into account `children.length` when calculating `extraAvatars`. Using the example from the `'should respect total'` test to illustrate the problem:
```jsx
<AvatarGroup total={10}>
  <Avatar src="/fake.png" />
  <Avatar src="/fake.png" />
  <Avatar src="/fake.png" />
</AvatarGroup>
```
In this example, there would be 10 `totalAvatars`, and since `max` was not specified, `clampedMax` would be 5. Applying the suggested calculation above, `extraAvatars` will equal 6 (10 - 5 + 1). However, this is incorrect, because there are only 3 child avatars available, so the component will render the 3 avatars and a "+6".